### ISSUE TITLE
Error in system2(inla.call, args = paste(all.args, shQuote(file.ini)),  :    7 arguments passed to .Internal(system) which requires 2 or 3

### ISSUE THREAD

**Issue by jung233** [hrue/r-inla#104]
When I tried to run, I ran into problems I couldn't resolve; any help is appreciated.
```
> data <- data.frame(y = c(0, 1, 0, 1), x = c(1, 2, 3, 4))
> result <- inla(y ~ x, data = data, family = "binomial")
Error in system2(inla.call, args = paste(all.args, shQuote(file.ini)),  : 
  7 arguments passed to .Internal(system) which requires 2 or 3

 *** inla.core.safe:  The inla program failed, but will rerun in case better initial values may help. try=1/1 
Error in system2(inla.call, args = paste(all.args, shQuote(file.ini)),  : 
  7 arguments passed to .Internal(system) which requires 2 or 3

Error in inla.core.safe(formula = formula, family = family, contrasts = contrasts,  (file_name.R#43): The inla program call crashed.
The inla program failed and the maximum number of tries has been reached.
```
```
> inla.version()
	R-INLA version ..........: 24.06.27
	Date ....................: 2024-06-27
	Maintainers .............: Havard Rue <hrue@r-inla.org>
	                         : Finn Lindgren <finn.lindgren@gmail.com>
	                         : Elias Teixeira Krainski <elias@r-inla.org>
	Main web-page ...........: www.r-inla.org
	Download-page ...........: inla.r-inla-download.org
	Repository ..............: github.com/hrue/r-inla
	Email support ...........: help@r-inla.org
	                         : r-inla-discussion-group@googlegroups.com
```

**Comment by hrue** [comment]
This sounds like an R-inconsistency thing, try with R-4.4 and install the most
recent testing version

On Mon, 2024-11-04 at 04:15 -0800, Raymond wrote:
> When I tried to run, I ran into problems I couldn't resolve; any help is
> appreciated.
> > data <- data.frame(y = c(0, 1, 0, 1), x = c(1, 2, 3, 4))
> > result <- inla(y ~ x, data = data, family = "binomial")
> Error in system2(inla.call, args = paste(all.args, shQuote(file.ini)),  : 
>   7 arguments passed to .Internal(system) which requires 2 or 3
> 
>  *** inla.core.safe:  The inla program failed, but will rerun in case better
> initial values may help. try=1/1 
> Error in system2(inla.call, args = paste(all.args, shQuote(file.ini)),  : 
>   7 arguments passed to .Internal(system) which requires 2 or 3
> 
> Error in inla.core.safe(formula = formula, family = family, contrasts =
> contrasts,  (file_name.R#43): The inla program call crashed.
> The inla program failed and the maximum number of tries has been reached.
> > inla.version()
> 	R-INLA version ..........: 24.06.27
> 	Date ....................: 2024-06-27
> 	Maintainers .............: Havard Rue ***@***.***>
> 	                         : Finn Lindgren ***@***.***>
> 	                         : Elias Teixeira Krainski ***@***.***>
> 	Main web-page ...........: www.r-inla.org
> 	Download-page ...........: inla.r-inla-download.org
> 	Repository ..............: github.com/hrue/r-inla
> 	Email support ...........: ***@***.***
> 	                         : ***@***.***
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID:
> ***@***.***>

-- 
Håvard Rue
Professor of Statistics
Chair of the Statistics Program
CEMSE Division
King Abdullah University of Science and Technology
Thuwal 23955-6900
Kingdom of Saudi Arabia

***@***.***
Office: +966 (0)12 808 0640  
Mobile: +966 (0)54 470 0421
Research group: bayescomp.kaust.edu.sa  
R-INLA project: www.r-inla.org
Zoom: kaust.zoom.us/my/haavard.rue

-- 

This message and its contents, including attachments are intended solely 
for the original recipient. If you are not the intended recipient or have 
received this message in error, please notify me immediately and delete 
this message from your computer system. Any unauthorized use or 
distribution is prohibited. Please consider the environment before printing 
this email.

**Comment by jung233** [comment]
Thanks for your help. I tried to reinstall R4.4.1 to 4.4.2, but found that the package could not be installed normally. It seemed that it was indeed a problem with R itself. After that, I changed the IDE from Dataspell to Rstudio and executed `install.packages("INLA",repos=c(getOption("repos"),INLA="https://inla.r-inla-download.org/R/testing"), dep=TRUE)`
Then the problem was solved. 
Thank you again！
